### ISSUE TITLE
Can't build with MinGW?

### ISSUE THREAD

**Issue by phated** [WebAssembly/binaryen#3039]
Recently, I've been trying to build Binaryen with MinGW to link into Grain's OCaml codebase, but the build fails halfway through with something like:
```
D:/a/binaryen.ml/binaryen.ml/src/binaryen/third_party/llvm-project/include/llvm/Object/SymbolicFile.h: In function 'OStream& llvm::object::operator<<(OStream&, const llvm::object::DataRefImpl&)':
D:/a/binaryen.ml/binaryen.ml/src/binaryen/third_party/llvm-project/include/llvm/Object/SymbolicFile.h:47:30: error: expected ')' before 'PRIxPTR'
   OS << "(" << format("0x%08" PRIxPTR, D.p) << " (" << format("0x%08x", D.d.a)
                      ~       ^~~~~~~~
                              )
```

Should Binaryen build with MinGW? I noticed some commits from 2016-2017 that mentioned support for MinGW, but it seems like it was removed from the CI matrix. Maybe I'm doing something wrong.

**Comment by gerald-dotcom** [comment]
Hit the same. 

Just edit cmake file and add following

add_compile_flag("-D__STDC_FORMAT_MACROS")

**Comment by phated** [comment]
@gerald-dotcom whoa! Thank you! Where exactly did you add that?

**Comment by gerald-dotcom** [comment]
On your text editor, Search for "add_compile_flag" and you will find a function named add_compile_flag then put this code under the function (Not inside).

**Comment by phated** [comment]
Fixed by #3287 - There is further work to be done to get the tests working, but at least it builds now.
